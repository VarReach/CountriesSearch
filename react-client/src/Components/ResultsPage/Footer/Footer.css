.results__footer {
  background-color: #141414;
  position: relative;
  bottom: 0;
  padding: 1em 2em 4em;
}

.results__footer__counts {
  display: flex;
}

.results__footer__regions,
.results__footer__subregions {
  width: 50%;
}

.results__footer__counts h3 {
  font-size: 0.95em;
  margin: 0;
  display: inline;
}
.results__footer__counts span {
  margin-right: 3em;
  display: block;
}