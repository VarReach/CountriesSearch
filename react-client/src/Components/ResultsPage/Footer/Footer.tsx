import React, { FC, useState, useEffect } from "react";
import "./Footer.css"
import { Country } from "../../../Services/CountriesService";

export interface FooterProps {
  countries: Country[],
}

export const Footer: FC<FooterProps> = (props) => {
  const { countries } = props;

  const [counts, setCounts] = useState<{ regionCounts, subregionCounts }>({ regionCounts: {}, subregionCounts: {} });

  useEffect(() => {
    // console.log(countries);
    const counts = countries.reduce((counts, country) => {
      const { region, subregion } = country;
      counts.regionCounts[region] 
        = counts.regionCounts[region] ? counts.regionCounts[region] + 1 : 1; 
      counts.subregionCounts[subregion] 
        = counts.subregionCounts[subregion] ? counts.subregionCounts[subregion] + 1 : 1; 
      return counts;
    }, { regionCounts: {}, subregionCounts: {} });
    setCounts(counts);
  }, [countries]);


  const renderCounts = (entries) => {
    return entries.map(([key, value], i) => {
      return <span key={i}><h3>{key ? key : "Unknown"}:</h3> {value}</span>
    })
  }

  const renderSubregionCounts = () => {
    const entries = Object.entries(counts.subregionCounts);
    if (entries.length) {
      return (
        <div className="results__footer__subregions">
          <h2>Subregions:</h2>
          {renderCounts(entries)}
        </div>
      );
    }
  }

  const renderRegionCounts = () => {
    const entries = Object.entries(counts.subregionCounts);
    if (entries.length) {
      return (
        <div className="results__footer__regions">
          <h2>Regions:</h2>
          {renderCounts(entries)}
        </div>
      );
    }
  }


  return (
    <div className="results__footer">
      <div className="results__footer__total-count">
        <h2>Total Number of Countries Found:</h2>
        <h3>{countries.length}</h3>
      </div>
      <div className="results__footer__counts">
        {renderRegionCounts()}
        {renderSubregionCounts()}
      </div>
    </div>
  );
}

export default Footer;