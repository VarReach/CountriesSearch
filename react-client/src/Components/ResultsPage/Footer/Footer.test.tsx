import React from "react";
import { render } from "@testing-library/react";
import mockCountries from "../../../../__mocks__/countriesMock";
// Components
import Footer, { FooterProps } from "./Footer";

/**
 * Covers tests for the ResultsPage Footer
 */

describe("<Footer/>", () => {
  const defaultProps: FooterProps = {
    countries: mockCountries,
  };

  test("renders without crashing", () => {
    const { container } = render(<Footer {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});