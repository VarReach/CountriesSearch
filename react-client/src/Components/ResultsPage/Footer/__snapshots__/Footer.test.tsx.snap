// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> renders without crashing 1`] = `
<div>
  <div
    class="results__footer"
  >
    <div
      class="results__footer__total-count"
    >
      <h2>
        Total Number of Countries Found:
      </h2>
      <h3>
        1
      </h3>
    </div>
    <div
      class="results__footer__counts"
    >
      <div
        class="results__footer__regions"
      >
        <h2>
          Regions:
        </h2>
        <span>
          <h3>
            Test-land
            :
          </h3>
           
          1
        </span>
      </div>
      <div
        class="results__footer__subregions"
      >
        <h2>
          Subregions:
        </h2>
        <span>
          <h3>
            Test-land
            :
          </h3>
           
          1
        </span>
      </div>
    </div>
  </div>
</div>
`;
